packages/backend/src/services/DemandeSejour.js

```diff
@@ -153,14 +153,16 @@ const query = {
       (DS.HEBERGEMENT -> 'hebergements')[0] ->> 'hebergementId' IN ('${hebergementIds.join("','")}') as "estInstructeurPrincipal"
     FROM front.demande_sejour ds
       JOIN front.organismes o ON o.id = ds.organisme_id
-    WHERE  (${getHebergementWhereQuery(hebergementIds)})
-       ${search.map((s) => ` AND ${s} `).join("")}
+    WHERE  statut <> 'BROUILLON'
+        AND (${getHebergementWhereQuery(hebergementIds)})
```

Comment:
Ceci doit être remplacé par :
`AND ds.hebergement #>> '{hebergements, 0, coordonnees, adresse, departement}' IN (${departement_codes.join(", ")})`
Pas besoin de faire une pré-requête pour récupérer les identifiants en amont.
Cela fait économiser une requête.

Reply:
En effet, mais il y a juste un pb dans ta requete, si la demande est liée a plusieur etablissement, le back user doit la voir meme si l'hebergement 0 n'et pas dans le bon departement.

Je propose de remplmacer par ca : 

```
departementIds
    .map(
      (d) => `JSONB_PATH_QUERY_ARRAY(
        HEBERGEMENT,
        '$.hebergements[*].coordonnees.adresse.departement'
      ) @> '["${d}"]'`,
    )
    .join(" OR ");
```